src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private final float loadFactor;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        this.buckets = new Node[initialCapacity];
+        this.size = 0;
+        this.loadFactor = loadFactor;
+    }
 
     @Override
     public void put(K key, V value) {
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
 
+        Node<K, V> current = buckets[bucketIndex];
+        Node<K, V> previous = null;
+
+        while (current != null) {
+            if (key == null && current.key == null
```

Comment:
All such conditions may be simplified
```suggestion
            if (key == current.key
```